src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,31 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
+import mate.academy.hibernate.relations.exceptions.DataNotFoundException;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import org.hibernate.SessionFactory;
+
+public class CountryServiceImpl extends AbstractService implements CountryService {
+    private final CountryDao countryDao;
+
+    public CountryServiceImpl(SessionFactory sessionFactory) {
+        super();
+        this.countryDao = new CountryDaoImpl(sessionFactory);
```

Comment:
Consider using dependency injection for the `CountryDao` instead of instantiating it directly in the constructor. This will improve the flexibility and testability of the code.